source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -108,8 +112,36 @@ void buildHeadersFromTable(Http::HeaderMap& headers, lua_State* state, int table
   }
 }
 
+// Initialize a request options by inspecting the provided table. Currently it sets "sampled" flag.
+// This also infers whether the request will be done asynchronously (fire-and-forget) or
+// synchronously.
+bool initializeRequestOptionsFromTable(Http::AsyncClient::RequestOptions& request_options,
```

Comment:
OK. Yeah, you caught me being lazy on this since "is_async" is not really a part of `request_options` per se (can't be passed to `Http::AsyncClient`). I'll try to refactor this.

Reply:
Refactored.